LayoutTests/storage/indexeddb/modern/resources/leak-1.js

```diff
@@ -1,80 +1,69 @@
 description("This tests that certain IDB object relationships don't cause leaks.");
 
-indexedDBTest(prepareDatabase);
+evalAndLog('dbname = "leak-1"');
+database = null;
 
-function log(message)
-{
-    debug(message);
+function prepareDatabase() {
+    return new Promise((resolve, reject) => {
+        let openRequest = indexedDB.open(dbname);
+        openRequest.onupgradeneeded = (event) => {
+            debug("Initial upgrade needed: Old version - " + event.oldVersion + " New version - " + event.newVersion);
+            database = openRequest.result;
+            versionChangeTransactionObserver = internals.observeGC(openRequest.transaction);
+            openRequestObserver = internals.observeGC(openRequest);
+            databaseObserver = internals.observeGC(database);
+            objectStoreObserver = internals.observeGC(database.createObjectStore("foo"));
+        }
+        openRequest.onsuccess = () => resolve();
+        openRequest.onerror = () => reject();
+    });
```

Comment:
It is usually better to have the least code in the Promise constructor.
I would tend to do:
```
const openRequest = ...
openRequest.onupgradeneeded = ...
return new Promise((resolve, reject) => {
    openRequest.onsuccess = resolve;
    openRequest.onerror = reject;
});
```